Assignment-1-Calculator(Updated).c

```diff
@@ -0,0 +1,167 @@
+    }
+
+    return 0;
+}
+
+/* Function to check the priority of the operator based on DMAS */
+int getOperatorPriority(char operator) {
+    if (operator == MULTIPLY || operator == DIVIDE) return 1;
+    if (operator == ADD || operator == SUBTRACT) return 0;
+    return -1;
+}
+
+/* Function to validate input for examples like `82*` or `3++4` */
+int validateInput(char input[]) {
+    int hasNumber = 0;
+    int hasOperator = 0;
+
+    for (int iteratorI = 0; input[iteratorI] != '\0'; iteratorI++) {
+        if (input[iteratorI] >= '0' && input[iteratorI] <= '9') {
+            hasNumber = 1;
+            hasOperator = 0;
+        } else if (input[iteratorI] == ADD || input[iteratorI] == SUBTRACT || input[iteratorI] == MULTIPLY || input[iteratorI] == DIVIDE) {
+            if (hasOperator && !(input[iteratorI] == SUBTRACT && (iteratorI == 0 || (input[iteratorI - 1] == ADD || input[iteratorI - 1] == SUBTRACT || input[iteratorI - 1] == MULTIPLY || input[iteratorI - 1] == DIVIDE)))) {
+                return 0;  // Consecutive operators are invalid except for leading negatives
+            }
+            hasOperator = 1;
+        } else {
+            return 0;  // Invalid character
+        }
+    }
+    return hasNumber && !hasOperator;  // Ensure the expression ends with a number
+}
+
+/* Function to remove blank spaces from input */
+void removeBlankSpaces(char input[]) {
+    int iteratorI = 0, index = 0;
+    while (input[iteratorI] != '\0') {
+        if (input[iteratorI] != ' ') {
+            input[index++] = input[iteratorI];
+        }
+        iteratorI++;
+    }
+    input[index] = '\0'; // Terminate cleaned string
+}
+
+/* Function which performs basic math operations */
+void calculateOperations(int numbers[], char operators[], int *numberIndex, int *operatorIndex, int *errorStatus) {
+    int operand1 = numbers[--(*numberIndex)];
+    int operand2 = numbers[--(*numberIndex)];
+    char operator = operators[(*operatorIndex)--];
+
+    if (operator == MULTIPLY) {
+        numbers[(*numberIndex)++] = operand2 * operand1;
+    } else if (operator == SUBTRACT) {
+        numbers[(*numberIndex)++] = operand2 - operand1;
+    } else if (operator == ADD) {
+        numbers[(*numberIndex)++] = operand2 + operand1;
+    } else if (operator == DIVIDE) {
+        if (operand1 == 0) {
+            *errorStatus = 2;
+            return;
+        }
+        numbers[(*numberIndex)++] = operand2 / operand1;
+    }
+
+    return;
+}
+
+/* Function to calculate remaining operations */
+int evaluateRemainingOperations(int numbers[], char operators[], int *numberIndex, int *operatorIndex, int *errorStatus) {
+    while ((*operatorIndex) != -1) {
+        calculateOperations(numbers, operators, numberIndex, operatorIndex, errorStatus);
+        if (*errorStatus) return 0;
+    }
+    return numbers[0];
+}
+
+/* Function to perform the calculation based on the input */
+int performCalculation(char input[], int *errorStatus) {

```

Comment:
can you cover  2*4/5*2 scenario, program should handle DMAS rule [division, multiplication, addition and subtraction]